pkg/srpmproc/process.go

```diff
@@ -1176,112 +1176,39 @@ func convertMetaData(pkgName string, localRepo string) bool {
-		log.Println(err)
-		return ""
-	}
-
-	// Loop through .<package>.metadata and get the file names we need to make our SRPM:
-	lookAside, err := os.Open(fmt.Sprintf("%s/.%s.metadata", localRepo, pkgName))
-	if err != nil {
-		log.Println(err)
-		return ""
+	if !strings.HasSuffix(specFile, ".spec") {
+		return "", fmt.Errorf("Error, first file found in SPECS/ is not a .spec file!  Check the SPECS/ directory in the repo?")
 	}
 
-	// Split file into lines and start processing:
-	scanner := bufio.NewScanner(lookAside)
-	scanner.Split(bufio.ScanLines)
-
-	// loop through each line, and:
-	//   - isolate the SOURCES/filename entry
-	//   - write out a dummy file of the same name to rpmBuildPath/SOURCES
-	for scanner.Scan() {
-
-		// lookaside source is always the 2nd part of the line (after the long SHA sum)
-		srcFile := strings.Fields(scanner.Text())[1]
-
-		// write a dummy file of the same name into the rpmbuild SOURCES/ directory:
-		dummyFile, err := os.OpenFile(fmt.Sprintf("%s/%s", rpmBuildPath, srcFile), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
-		if err != nil {
-			return ""
-		}
-		writer := bufio.NewWriter(dummyFile)
-		_, _ = writer.WriteString("This is a dummy lookaside file generated by srpmproc.  It is only needed to get a working SRPM and extract version information.  Please disregard\n")
-		writer.Flush()
-		dummyFile.Close()
-	}
-
-	lookAside.Close()
-
-	// Now, call rpmbuild to produce the dummy src file:
-	// Example:  rpmbuild  --define "_topdir  /tmp/srpmproctmp_httpd1988142783_rpm"  -bs /tmp/srpmproctmp_httpd1988142783_rpm/SPECS/*.spec
-	cmd := exec.Command("rpmbuild", fmt.Sprintf(`--define=_topdir  %s`, rpmBuildPath), fmt.Sprintf(`--define=dist  .el%d`, majorVersion), "-bs", fmt.Sprintf("%s/SPECS/%s.spec", rpmBuildPath, pkgName))
-	if err := cmd.Run(); err != nil {
-		log.Println(err)
-		return ""
-	}
-
-	// Read the first file from the SRPMS/ folder in rpmBuildPath.  It should be the SRPM that rpmbuild produced above
-	// (there should only be one file - we check that it ends in ".rpm" just to be sure!)
-	lsTmp, err := ioutil.ReadDir(fmt.Sprintf("%s/SRPMS/", rpmBuildPath))
-	if err != nil {
-		log.Println(err)
-		return ""
-	}
-
-	srpmFile := lsTmp[0].Name()
-
-	if !strings.HasSuffix(srpmFile, ".rpm") {
-		log.Println("Error, file found in dummy SRPMS directory did not have an .rpm extension!  Perhaps rpmbuild didn't produce a proper source RPM?")
-		return ""
-	}
-
-	// Call the rpm binary to extract the version-release info out of it, and tack on ".el<VERSION>" at the end:
-	cmd = exec.Command("rpm", "-qp", "--qf", `%{NAME}|%{VERSION}|%{RELEASE}\n`, fmt.Sprintf("%s/SRPMS/%s", rpmBuildPath, srpmFile))
+	// Call the rpmspec binary to extract the version-release info out of it, and tack on ".el<VERSION>" at the end:
+	cmd := exec.Command("rpmspec", "--srpm", fmt.Sprintf(`--define=dist  .el%d`, majorVersion), "-q", "--queryformat", `%{NAME}|%{VERSION}|%{RELEASE}\n`, fmt.Sprintf("%s/SPECS/%s", localRepo, specFile))
```

Comment:
small nitpick, can you split the args to the command into a variable and pass them to exec.Command()?

e.g.

```
	cmdArgs := []string{
		"--srpm",
		fmt.Sprintf(`--define=dist  .el%d`, majorVersion),
		"-q",
		"--queryformat",
		`%{NAME}|%{VERSION}|%{RELEASE}\n`,
		fmt.Sprintf("%s/SPECS/%s", localRepo, specFile),
	}
	cmd := exec.Command("rpmspec", cmdArgs...)

```